bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/webhook/NetatmoServlet.java

```diff
@@ -0,0 +1,159 @@
+import org.openhab.binding.netatmo.internal.api.SecurityApi;
+import org.openhab.binding.netatmo.internal.api.dto.WebhookEvent;
+import org.openhab.binding.netatmo.internal.deserialization.NADeserializer;
+import org.openhab.binding.netatmo.internal.handler.ApiBridgeHandler;
+import org.openhab.binding.netatmo.internal.handler.capability.EventCapability;
+import org.osgi.service.http.HttpService;
+import org.osgi.service.http.NamespaceException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * HTTP servlet for Netatmo Webhook.
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class NetatmoServlet extends HttpServlet {
+    private static final long serialVersionUID = -354583910860541214L;
+    private static final String CALLBACK_URI = "/" + BINDING_ID;
+
+    private final Logger logger = LoggerFactory.getLogger(NetatmoServlet.class);
+    private final Map<String, EventCapability> dataListeners = new ConcurrentHashMap<>();
+    private final HttpService httpService;
+    private final NADeserializer deserializer;
+    private final Optional<SecurityApi> securityApi;
+    private boolean hookSet = false;
+
+    public NetatmoServlet(HttpService httpService, ApiBridgeHandler apiBridge, String webHookUrl) {
+        this.httpService = httpService;
+        this.deserializer = apiBridge.getDeserializer();
+        this.securityApi = Optional.ofNullable(apiBridge.getRestManager(SecurityApi.class));
+        securityApi.ifPresent(api -> {
+            try {
+                httpService.registerServlet(CALLBACK_URI, this, null, httpService.createDefaultHttpContext());
+                logger.debug("Started Netatmo Webhook Servlet at '{}'", CALLBACK_URI);
+                URI uri = UriBuilder.fromUri(webHookUrl).path(BINDING_ID).build();
+                try {
+                    logger.info("Setting Netatmo Welcome WebHook to {}", uri.toString());
+                    api.addwebhook(uri);
+                    hookSet = true;
+                } catch (UriBuilderException e) {
+                    logger.info("webhookUrl is not a valid URI '{}' : {}", uri, e.getMessage());
+                } catch (NetatmoException e) {
+                    logger.info("Error setting webhook : {}", e.getMessage());
+                }
+            } catch (ServletException | NamespaceException e) {
+                logger.warn("Could not start Netatmo Webhook Servlet : {}", e.getMessage());
+            }
+        });
+    }
+
+    public void dispose() {
+        securityApi.ifPresent(api -> {
+            if (hookSet) {
+                logger.info("Releasing Netatmo Welcome WebHook");
+                try {
+                    api.dropWebhook();
+                } catch (NetatmoException e) {
+                    logger.warn("Error releasing webhook : {}", e.getMessage());
+                }
+            }
+        });
+        httpService.unregister(CALLBACK_URI);
```

Comment:
As registerServlet is done only when securityApi is present, I would suggest to call unregister when securityApi is present.

Reply:
Changed